src/modelmanager.cpp

```diff
@@ -225,6 +226,62 @@ Status ModelManager::loadPipelinesConfig(rapidjson::Document& configJson) {
     return ovms::StatusCode::OK;
 }
 
+Status ModelManager::loadCustomLoadersConfig(rapidjson::Document& configJson) {
+    const auto itrp = configJson.FindMember("custom_loader_config_list");
+    if (itrp == configJson.MemberEnd() || !itrp->value.IsArray()) {
+        return StatusCode::OK;
+    }
+
+    // Load Customer Loaders as per the configuration
+    SPDLOG_INFO("Using Customloaders");
+    auto& customloaders = ovms::CustomLoaders::instance();
+    for (const auto& configs : itrp->value.GetArray()) {
+        const std::string loaderName = configs["config"]["loader_name"].GetString();
+        SPDLOG_INFO("Reading Custom Loader:{} configuration", loaderName);
+
+        CustomLoaderConfig loaderConfig;
+        auto status = loaderConfig.parseNode(configs["config"]);
+        if (status != StatusCode::OK) {
+            SPDLOG_ERROR("Parsing loader:{} config failed", loaderName);
+            return status;
+        }
+
+        SPDLOG_INFO("CHECK IF LOADER IS ALREADY LOADED");
+        if (customloaders.find(loaderName) == nullptr) {
+            // this is where library or custom loader is loaded
+            void* handleCL = dlopen(const_cast<char*>(loaderConfig.getLibraryPath().c_str()), RTLD_LAZY | RTLD_LOCAL);
+            if (!handleCL) {
+                SPDLOG_ERROR("Cannot open library:  {} {}", loaderConfig.getLibraryPath(), dlerror());
+                return StatusCode::CUSTOM_LOADER_LIBRARY_INVALID;
+            }
+
+            // load the symbols
+            createCustomLoader_t* customObj = (createCustomLoader_t*)dlsym(handleCL, "createCustomLoader");
+            const char* dlsym_error = dlerror();
+            if (dlsym_error) {
+                SPDLOG_ERROR("Cannot load symbol create:  {} ", dlsym_error);
+                return StatusCode::CUSTOM_LOADER_LIBRARY_LOAD_FAILED;
+            }
+
+            std::shared_ptr<CustomLoaderInterface> customerLoaderIfPtr{customObj()};
+            try {
+                customerLoaderIfPtr->loaderInit(loaderConfig.getLoaderConfigFile());
+            } catch (...) {
+                SPDLOG_ERROR("Cannot create or initialize the custom loader");
+                return StatusCode::CUSTOM_LOADER_INIT_FAILED;
+            }
```

Comment:
It would make sense here to at least try first to catch std::exception, print error message if any and only if it fails catch with "...".

Reply:
Added catch for std::exception